micov/cli.py

```diff
@@ -534,48 +553,19 @@ def binning(
         .sort("sample_hits_std", descending=True)
     )
 
-    df_bins_by_read_hits = (
-        df_bins.group_by(COLUMN_GENOME_ID, "bin_idx", "bin_start", "bin_stop")
-        .agg(pl.col("read_hits").std().alias("read_hits_std"))
-        .fill_null(0)
-        .sort("read_hits_std", descending=True)
-    )
-
-    df_bins = df_bins.with_columns(
-        [
-            pl.col("bin_start").cast(COLUMN_START_DTYPE),
-            pl.col("bin_stop").cast(COLUMN_STOP_DTYPE),
-        ]
-    )
-    df_bins_by_sample_hits = df_bins_by_sample_hits.with_columns(
-        [
-            pl.col("bin_start").cast(COLUMN_START_DTYPE),
-            pl.col("bin_stop").cast(COLUMN_STOP_DTYPE),
-        ]
-    )
-    df_bins_by_read_hits = df_bins_by_read_hits.with_columns(
-        [
-            pl.col("bin_start").cast(COLUMN_START_DTYPE),
-            pl.col("bin_stop").cast(COLUMN_STOP_DTYPE),
-        ]
-    )
-
     os.makedirs(outdir, exist_ok=True)
-    make_csv_ready(df_bins).collect().write_csv(
+    make_csv_ready(df_bins).sink_csv(
         f"{outdir}/stats_bins.tsv",
         separator="\t",
         include_header=True,
     )
+
+    # it's not obvious but this cannot use sink_csv?'
     df_bins_by_sample_hits.collect().write_csv(
```

Comment:
maybe we can output the top N rows of `df_bins_by_sample_hits` since we are only interested in the top ranked regions

Reply:
nah not hard to post process but no reason to output things which we think are uninteresting